kernel/src/fs/utils/xattr.rs

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use crate::prelude::*;
+
+pub const XATTR_NAME_MAX_LEN: usize = 255;
+pub const XATTR_VALUE_MAX_LEN: usize = 65536;
+pub const XATTR_LIST_MAX_LEN: usize = 65536;
+
+/// Represents different namespaces with different capabilities
+/// for extended attributes (xattrs).
+#[derive(Debug, PartialEq, Eq, Clone, Copy, TryFromInt, Hash)]
+#[repr(u8)]
+pub enum XattrNamespace {
+    User = 1,
+    Trusted = 2,
+    System = 3,
+    Security = 4,
+    // More namespaces can be added here.
+}
+
+#[derive(Debug, Hash)]
+pub struct XattrName<'a> {
+    namespace: XattrNamespace,
+    name_str: &'a str,
```

Comment:
It is unclear if `name_str` contains the namespace prefix. Since the namespace info is already encoded, it is natural for one to expect that the `name_str` has been stripped of the namespace prefix. But this is not the case in the current implementation.

Which one is a better design decision: strip `name_str` of the prefix or not?  Either way, we need to make this point more obvious through better naming and comments.